Upgrade EUI to v95.11.0

cee-chen opened the pull request:
`v95.10.1`⏩`v95.11.0`

> [!note]
> The bulk of this release is **EuiDataGrid**. The component has been fully converted to Emotion, and several UX changes have been made to data cell actions. We recommend QA testing any data grid(s) that have custom styles applied to them.

_[Questions? Please see our Kibana upgrade FAQ.](https://github.com/elastic/eui/blob/main/wiki/eui-team-processes/upgrading-kibana.md#faq-for-kibana-teams)_

---

## [`v95.11.0`](https://github.com/elastic/eui/releases/v95.11.0)

- Updated `EuiText`, `EuiTextColor`, and `EuiTextAlign` with a new `component` prop that allows changing the default rendered `<div>` wrapper to a `<span>` or `<p>` tag. ([#7993](https://github.com/elastic/eui/pull/7993))
- Updated `EuiDataGrid`'s cell actions to always consistently be left-aligned, regardless of text content alignment ([#8011](https://github.com/elastic/eui/pull/8011))
- Increased `EuiDataGrid`'s cell actions hover zone to reduce UX friction when mousing over from the grid cell to its actions ([#8011](https://github.com/elastic/eui/pull/8011))

**Bug fixes**

- Fixed `EuiPopover` to correctly inherit from `EuiProvider`'s `componentDefaults.EuiPortal.insert` ([#8003](https://github.com/elastic/eui/pull/8003))
- Fixed push `EuiFlyoutResizable`s to not potentially block scrollbars on outside content ([#8010](https://github.com/elastic/eui/pull/8010))
- Fixed an `EuiDataGrid` bug where the `setCellProps` callback passed by `renderCellValue` was not correctly applying custom `data-test-subj`s ([#8011](https://github.com/elastic/eui/pull/8011))

**Accessibility**

- Updated the `EuiBasicTable` actions button's `aria-label` by adding a reference to the current row ([#7994](https://github.com/elastic/eui/pull/7994))

**CSS-in-JS conversions**

- Converted `EuiDataGrid`'s toolbar controls to Emotion ([#7997](https://github.com/elastic/eui/pull/7997))
  - Removed `$euiDataGridPopoverMaxHeight`
- Converted `EuiDataGrid` to Emotion ([#7998](https://github.com/elastic/eui/pull/7998))
  - Removed `$euiZDataGrid`
  - Removed `$euiZHeaderBelowDataGrid`
- Converted `EuiDataGrid`'s `gridStyle`s to Emotion; Removed the following Sass variables and mixins: ([#8006](https://github.com/elastic/eui/pull/8006))
  - `$euiDataGridCellPaddingS`
  - `$euiDataGridCellPaddingM`
  - `$euiDataGridCellPaddingL`
  - `$euiDataGridVerticalBorder`
  - `$euiDataGridPrefix`
  - `$euiDataGridStyles`
  - `@euiDataGridSelector`
  - `@euiDataGridStyles`
- Converted `EuiDataGrid`'s cell popover, actions, and focus outline to Emotion; Removed the following Sass variables and mixins: ([#8011](https://github.com/elastic/eui/pull/8011))
  - `$euiZDataGridCellPopover`
  - `@euiDataGridCellFocus`
- Converted `EuiDataGrid`'s row, header, and footer cells to Emotion; Removed the following Sass variables and mixins: ([#8013](https://github.com/elastic/eui/pull/8013))
  - `$euiDataGridColumnResizerWidth`
  - `@euiDataGridRowCell`
  - `@euiDataGridHeaderCell`
  - `@euiDataGridFooterCell`


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

cee-chen committed:
Upgrade EUI to v95.11.0

cee-chen committed:
i18n updates

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!

cee-chen committed:
[FTR] Update EuiDataGrid service based on DOM cleanup

- header control cells no longer have `.euiDataGridHeaderCell__content` wrappers

cee-chen committed:
[Discover] Clean up EuiDataGrid styles to account for new styles

- `.euiDataGridHeaderCell__button` now behaves very differently due to interactive children change, so center it instead of stretching it

- `.euiDataGridHeaderCell__popover` no longer exists

- `.euiDataGridRowCell__content` no longer exists for control columns

- `.euiDataGridRowCell--numeric` text alignment is already being applied by EUI, delete it

cee-chen committed:
Fix Enzyme tests to account for new Emotion wrappers

cee-chen committed:
Update datagrid cell content test assertion w/ removed screen reader text

cee-chen committed:
Fix RTL tests with updated DOM

- `.euiDataGridHeaderCell__icon` no longer exists, use the data-test-subj attached directly to the column actions button instead

- `.euiDataGridHeaderCell__content` no longer exists in control columns

- `Unsort` text has been added to datagrid column actions to indicate a column is already being sorted

cee-chen committed:
Update EuiBasicTable to account for new a11y text

- row # has been added to each actions button

cee-chen committed:
[ML][FTR] Fix test selector based on removed SR text

cee-chen committed:
[FTR] Harden datagrid cell actions clicking

- not totally sure how the Emotion conversion broke this :T but hovering over cells/etc helps ensure the button remains clickable

cee-chen committed:
[ML][FTR] Fix test to account for removed className

- prefer className specific to column selector instead

cee-chen committed:
[Discover][FTR] Update cell headers assertion to account for screen reader text

peter_griffin_blinds.gif

cee-chen committed:
[Lens][FTR] Fix FTR sorting failure

- `.euiDataGridHeader__action--selected` isn't a className that EUI sets anymore (as it wasn't doing anything) - we use sort/unsort copy

- Lens is overriding EUI's copy, so we need to update the tests to be more manual and remove the `none` option

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

justinkambic reviewed (approved):
SLO changes LGTM

afharo reviewed (approved):
Core changes LGTM

jughosta commented:
Looks like the default `EuiDataGrid` header styles got changed or how can we avoid visual changes now?

Before:
<img width="1137" alt="Screenshot 2024-09-18 at 14 32 23" src="https://github.com/user-attachments/assets/d679dbf2-f7f2-4502-b6eb-cf1045cfbf6c">

After (notice a top border above the grid header):
<img width="1013" alt="Screenshot 2024-09-18 at 14 33 40" src="https://github.com/user-attachments/assets/daee64fa-4310-48e4-84e3-3d90f7d81984">

Before:
<img width="537" alt="Screenshot 2024-09-18 at 14 34 15" src="https://github.com/user-attachments/assets/5a27bcff-bdad-4036-b346-1126f2f7d2fe">

After (notice a top border above the grid header and a thick bottom border):
<img width="573" alt="Screenshot 2024-09-18 at 14 33 57" src="https://github.com/user-attachments/assets/9e72a407-f31c-4c8e-9d4f-7a9f813e6b25">

cee-chen committed:
Merge branch 'main' into eui/v95.11.0

cee-chen committed:
[PR feedback] Restore Discover custom header borders

cee-chen commented:
@jughosta The discover data grids should be fixed now (f8cae9d) to the "before" screenshots in your comment - let me know if not, or if you see any other areas that need fixes!

cee-chen committed:
Merge branch 'main' into eui/v95.11.0

peteharverson reviewed (approved):
ML changes LGTM. Tested locally (data grid usage in transforms and DFA).

jughosta commented on packages/kbn-unified-data-table/src/components/data_table.scss:
```diff
@@ -78,7 +81,7 @@
     border-right: 0;
   }
-  .euiDataGridRowCell.euiDataGridRowCell--controlColumn[data-gridcell-column-id='colorIndicator'] .euiDataGridRowCell__content {
+  .euiDataGridRowCell.euiDataGridRowCell--controlColumn[data-gridcell-column-id='colorIndicator'] {
```
We added row color indicators in https://github.com/elastic/kibana/pull/192756/files and in this PR they don't show up any more. Looks like it has something to do with the cell height change. Would it be possible to fix?

cee-chen replied:
Thank you so much for catching this Julia, I was overly aggressive in updating some CSS selectors 🤦 This should be fixed now! Please let me know if not and if your eagle eyes spot anything else!

l-suarez reviewed (commented):
From the design point of view, all looks good. 

I checked for visual regressions comparing current state and this pull request and didn't find differences. DataGrids checked in **both** light and dark theme:
- Dashboards
- Lens
- Discover

Thank you!

PS - @elastic/kibana-design could you double check at your end? I'm not 100% familiar with the product yet and I might be missing something :S

cee-chen committed:
[PR feedback] Revert incorrectly removed styles targeting row cell content div

I was derping and should not have removed the `__content` selector :facepalm:

PhilippeOberti reviewed (approved):
code LGTM for the Threat Hunting Investigations team!

jughosta reviewed (approved):
Data Discovery changes LGTM, thanks! 👍

markov00 commented on x-pack/plugins/lens/public/visualizations/datatable/components/table_basic.test.tsx:
```diff
@@ -141,9 +141,9 @@ describe('DatatableComponent', () => {
     expect(screen.getByLabelText('My fanci metric chart')).toBeInTheDocument();
     expect(screen.getByRole('row')).toBeInTheDocument();
     expect(screen.queryAllByRole('gridcell').map((cell) => cell.textContent)).toEqual([
-      'shoes- a, column 1, row 1',
-      '1588024800000- b, column 2, row 1',
-      '3- c, column 3, row 1',
```
I'm wondering why this has changed

cee-chen replied:
Follow up note: if you still wish to assert on cell col/row placement in EuiDataGrid, please use the `data-gridcell-column-index` and `data-gridcell-visible-row-index` attributes.

markov00 commented on x-pack/test/functional/apps/lens/group2/table_dashboard.ts:
```diff
@@ -28,7 +28,7 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
     expect(await lens.getDatatableCellText(0, 0)).to.eql('169.228.188.120');
     // Remove the sorting
     await retry.try(async () => {
-      await lens.changeTableSortingBy(0, 'none');
```
There is no more 'none' sorting?

cee-chen replied:
It looks like there's a couple duplicate questions around this - let's consolidate to one thread. Please see https://github.com/elastic/kibana/pull/192756/commits/4eabb39e05cac7f5960230341ba978761cc2ed8d and https://github.com/elastic/kibana/pull/192756#discussion_r1768637396

dej611 commented on x-pack/test/functional/apps/lens/group2/table.ts:
```diff
@@ -36,7 +36,7 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       // Remove the sorting
       await retry.try(async () => {
-        await lens.changeTableSortingBy(0, 'none');
+        await lens.changeTableSortingBy(0, 'descending');
```
Why this change?

dej611 replied:
Had a look at the code underneath this change: apparently the new datagrid does mark any more the current sorting button as selected on the column actions dropdown.
So right now there's no way to get which "sorting" mode is currently "active" other than looking at the presence of the icon on the column.
Before a `--selected` class was added to the item.

Is this intentional?

cee-chen replied:
Yep, I removed `.euiDataGridHeader__action--selected` from our classNames/CSS as it wasn't actually applying any useful styles anymore, and I hadn't realized this FTR test was hooking into it 😬 I replaced it instead with logic that adds "Sort/Unsort" text to represent currently sorted state:

<img width="155" alt="" src="https://github.com/user-attachments/assets/b613c51d-4fe5-4abd-97ba-d5e4fcb8ba50">

But it looks like the Lens table completely overrides the sort label with your own `Sort ascending/descending` node, which means that the default "Unsort" text can't be hooked into.

Options I can look into:

- Remove your `label` override with a `schemaLabel` instead, which means that you get EuiDataGrid's "Unsort" text (the `schemaLabel` replaces the `A-Z/Z-A` text only)
- Re-add a selected state as a `data-` attribute of some kind that you can hook into (would likely be the next release, and I can revert the Lens changes at that point)

dej611 commented on x-pack/test/functional/apps/lens/group2/table.ts:
```diff
@@ -73,7 +73,7 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
       await retry.try(async () => {
-        await lens.changeTableSortingBy(4, 'none');
+        await lens.changeTableSortingBy(4, 'descending');
```
Why this change?

cee-chen replied:
It looks like there's a couple duplicate questions around this - let's consolidate to one thread. Please see https://github.com/elastic/kibana/pull/192756/commits/4eabb39e05cac7f5960230341ba978761cc2ed8d and https://github.com/elastic/kibana/pull/192756#discussion_r1768637396

ryankeairns commented:
I don't know for certain it was this PR that changed it, but the header is no longer fully clickable.
Perhaps that was intentional, so I ask to clarify.

For comparison, on the oblt demo instance (8.16 snapshot), you can click anywhere in the column header to open the action menu (sort, etc.). In this PR, you now must click on the overflow menu button to open the menu. As I said, this may well have been intentional but want to double check the source/reason for the change.

Thanks!

cee-chen commented:
@ryankeairns This change was not made in this EUI release or upgrade, it was made in https://github.com/elastic/kibana/pull/192026. Please see https://github.com/elastic/eui/pull/7898, https://github.com/elastic/eui/issues/7660, and https://github.com/elastic/eui/issues/7469 for more context on the change and why several Kibana teams have requested it.

ryankeairns reviewed (approved):
Thanks for the additional context!

cee-chen committed:
Merge branch 'main' into eui/v95.11.0

cee-chen commented:
@dej611 @markov00 I'm moving forward with asking KibanaOps for an admin merge as we need to unblock EUI's release/upgrade process. @elastic/kibana-visualizations is the only CODEOWNER team remaining and your owned code only touches tests and no actual source code/user-impacting code, so I feel comfortable getting this into main for now. If you'd like me to follow up with another PR later, please feel free to ping me separately or over Slack. Thanks!

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/193803">](https://github.com/elastic/kibana/pull/193803)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
